src/Microsoft.ServiceFabric.Services.Remoting/V2/ActivityIdLogicalCallContext.cs

```diff
@@ -0,0 +1,111 @@
+    /// Class ActivityIdLogicalCallContext.
+    /// </summary>
+    internal class ActivityIdLogicalCallContext
+    {
+        /// <summary>
+        /// Creates the activity.
+        /// Note : You still need to start it.
+        /// </summary>
+        /// <param name="activityMessage">The activity message.</param>
+        /// <returns>Activity.</returns>
+        internal Activity CreateW3CActivity(string activityMessage = "Create new Activity")
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+            var activity = new Activity(activityMessage);
+            return activity;
+        }
+
+        internal Activity StartActivity(IServiceRemotingRequestMessage requestMessage, string activityMessage = "Start new Activity")
+        {
+            var activity = new Activity(activityMessage);
+            string parentId = null;
+            if (requestMessage.GetHeader().ActivityIdParent != null)
+            {
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                parentId = requestMessage.GetHeader().ActivityIdParent;
+            }
+            else
+            {
+                parentId = requestMessage.GetHeader().ActivityRequestId;
+            }
+
+            if (!string.IsNullOrEmpty(parentId))
+            {
+                activity.SetParentId(parentId);
+                if (requestMessage.GetHeader().ActivityIdTraceStateHeader != null)
+                {
+                    activity.TraceStateString = requestMessage.GetHeader().ActivityIdTraceStateHeader;
+                }
+            }
+            else
+            {
+                Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+                Activity.ForceDefaultIdFormat = true;
+                activity.Start();
+                return activity;
+            }
+
+            // We expect baggage to be empty by default
+            // Only very advanced users will be using it in near future, we encourage them to keep baggage small (few items)
+            if (requestMessage.GetHeader().ActivityIdBaggage != null)
+            {
+                var baggage = requestMessage.GetHeader().ActivityIdBaggage;
+                foreach (var item in baggage)
+                {
+                    activity.AddBaggage(item.Key, item.Value);
+                }
+            }
+
+            activity.Start();
+            return activity;
+        }
+
+        /// <summary>
+        /// Determines whether this instance is present.
+        /// </summary>
+        /// <returns><c>true</c> if this instance is present; otherwise, <c>false</c>.</returns>
+        internal bool IsPresent()
+        {
+            return Activity.Current != null;
+        }
+
+        /// <summary>
+        /// Tries the get.
+        /// </summary>
+        /// <returns><c>true</c> if XXXX, <c>false</c> otherwise.</returns>

```

Comment:
Not true

Reply:
Have an IsPresent method. So changed this to Get()